theano/gpuarray/opt.py

```diff
@@ -994,19 +994,12 @@ def local_gpua_advanced_incsubtensor(op, context_name, inputs, outputs):
 
     x, y, ilist = inputs
 
-    # Gpu Ops needs both inputs to have the same dtype
-    if (x.type.dtype != y.type.dtype):
-        dtype = scalar.upcast(x.type.dtype, y.type.dtype)
-        if x.type.dtype != dtype:
-            x = tensor.cast(x, dtype)
-        if y.type.dtype != dtype:
-            y = tensor.cast(y, dtype)
-
     set_instead_of_inc = op.set_instead_of_inc
 
     compute_capability = int(context.bin_id[-2])
 
-    if (compute_capability < 2 or x.ndim != 2 or y.ndim != 2):
+    if (compute_capability < 2 or x.ndim != 2 or y.ndim != 2 or
+        x.type.dtype != y.type.dtype):
         return GpuAdvancedIncSubtensor1(
```

Comment:
The cast works, but is unused.  The optimization just returns the op, not a full node.  I figure this was done during the move to GraphToGpu and since we don't actually need the case for the non-dev20 case, I just removed it.